syntax = "proto3";

package bcdice_irc_proto;

service BCDiceInfoService {
  // GetBCDiceVersionInfo はBCDice IRCおよび関連プログラムのバージョン情報を返す。
  rpc GetBCDiceVersionInfo(GetBCDiceVersionInfoRequest) returns (GetBCDiceVersionInfoResponse);
  // GetDiceBotList は、ダイスボットの一覧を返す。
  rpc GetDiceBotList(GetDiceBotListRequest) returns (GetDiceBotListResponse);
  // Stop はサービスを停止する。
  rpc Stop(StopRequest) returns (StopResponse);
}

message GetBCDiceVersionInfoRequest {}

// GetBCDiceVersionInfoResponse は、BCDice IRCおよび関連プログラムのバージョン情報を含む応答。
message GetBCDiceVersionInfoResponse {
  // bcdice_version はBCDiceのバージョンを示す。
  string bcdice_version = 1;
  // bcdice_irc_version はBCDice IRCのバージョンを示す。
  string bcdice_irc_version = 2;
}

message GetDiceBotListRequest {}

// GetDiceBotListResponse は、ダイスボットの一覧を含む応答。
message GetDiceBotListResponse {
  // dice_bots はダイスボットの一覧。
  repeated DiceBot dice_bots = 1;
}

// DiceBot はダイスボットの情報を示す。
message DiceBot {
  // id はゲームシステムID。
  string id = 1;
  // name はゲームシステム名。
  string name = 2;
  // help_message は使い方の説明文。
  string help_message = 3;
}

message StopRequest {}

// StopResponse はサービス停止についての応答。
message StopResponse {
  // ok は正常にサービスを終了できたか。
  bool ok = 1;
}
